api/proto/teleport/notifications/v1/notifications_service.proto

```diff
@@ -0,0 +1,189 @@
+/*
+ * Teleport
+ * Copyright (C) 2023  Gravitational, Inc.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+syntax = "proto3";
+
+package teleport.notifications.v1;
+
+import "google/protobuf/empty.proto";
+import "teleport/notifications/v1/notifications.proto";
+
+option go_package = "github.com/gravitational/teleport/api/gen/proto/go/teleport/notifications/v1;notificationsv1";
+
+// NotificationService provides CRUD operations for notifications resources.
+service NotificationService {
+  // CreateUserNotification creates a user-specific notification.
+  rpc CreateUserNotification(CreateUserNotificationRequest) returns (Notification);
+  // DeleteUserNotification deletes a user-specific notification.
+  rpc DeleteUserNotification(DeleteUserNotificationRequest) returns (google.protobuf.Empty);
+  // ListUserNotifications returns a page of user-specific notifications.
+  rpc ListUserNotifications(ListUserNotificationsRequest) returns (ListUserNotificationsResponse);
+
+  // CreateGlobalNotification creates a global notification.
+  rpc CreateGlobalNotification(CreateGlobalNotificationRequest) returns (GlobalNotification);
+  // DeleteGlobalNotification deletes a global notification.
+  rpc DeleteGlobalNotification(DeleteGlobalNotificationRequest) returns (google.protobuf.Empty);
+  // ListGlobalNotifications returns a page of global notifications.
+  rpc ListGlobalNotifications(ListGlobalNotificationsRequest) returns (ListGlobalNotificationsResponse);
+
+  // CreateUserNotificationState creates a user notification state.
+  rpc CreateUserNotificationState(CreateUserNotificationStateRequest) returns (UserNotificationState);
+  // UpdateUserNotificationState updates a user notification state.
+  rpc UpdateUserNotificationState(UpdateUserNotificationStateRequest) returns (UserNotificationState);
+  // DeleteUserNotificationState deltes a user notification state.
+  rpc DeleteUserNotificationState(DeleteUserNotificationStateRequest) returns (google.protobuf.Empty);
```

Comment:
I think it would make sense to in order to minimize backend clutter, the only concern that comes to mind is handling SSO users since they technically only exist while logged in. 

When an SSO user logs out, are we _deleting_ the user the same way we'd delete a local user with `tctl`? If so, we'd need to handle that case separately and not delete their state objects since we want their notification states to persist between sessions. We could then rely on a periodic cleanup that deletes all notification state objects older than 30 days old to clean their state objects up.